subprojects/core/src/main/java/org/gradle/api/internal/attributes/DefaultMutableAttributeContainer.java

```diff
@@ -20,22 +20,27 @@
 import com.google.common.collect.Sets;
 import org.gradle.api.attributes.Attribute;
 import org.gradle.api.attributes.AttributeContainer;
+import org.gradle.api.internal.provider.ProviderInternal;
+import org.gradle.api.provider.Provider;
 import org.gradle.internal.Cast;
 
+import javax.annotation.Nullable;
 import java.util.Map;
+import java.util.Objects;
 import java.util.Set;
 
 class DefaultMutableAttributeContainer implements AttributeContainerInternal {
-    private final ImmutableAttributesFactory cache;
+    private final ImmutableAttributesFactory immutableAttributesFactory;
     private final AttributeContainerInternal parent;
     private ImmutableAttributes state = ImmutableAttributes.EMPTY;
+    private final Map<Attribute<?>, Provider<?>> lazyAttributes = Maps.newHashMap();
```

Comment:
Please use `new HashMap<>();` for the RHS. I suspect this should also remove the need to use a cast in the `forEach` use below.

From the javadoc of `Maps#newHashMap()`:
```
Note for Java 7 and later: this method is now unnecessary and should be treated as deprecated. Instead, use the HashMap constructor directly, taking advantage of the new "diamond" syntax .
```

Even without the "soft" deprecation above, I think we should not add any more usages of guava.